bindings/rust/s2n-tls-tokio/tests/common/mod.rs

```diff
@@ -67,14 +67,16 @@ pub fn server_config() -> Result<config::Builder, Error> {
 
 pub fn client_config_tls12() -> Result<config::Builder, Error> {
     let mut builder = config::Config::builder();
-    builder.set_security_policy(&DEFAULT)?;
+    let policy = Policy::from_version("20240501").unwrap();
+    builder.set_security_policy(&policy)?;
```

Comment:
I'm not convinced that the bindings tests should be frozen like the unit tests. They're arguably higher level tests and and should probably run with whatever the underlying C library defines as the default.

Reply:
This function promises to return a config that will negotiate TLS1.2 so I think we do want to pin to a max TLS1.2 security policy here.

If I change this to DEFAULT_TLS13, then the `shutdown_after_split` test starts to fail (timeout).

Reply:
I think you might be right about the grep_simple_mistakes as it currently says, "Tests should use a numbered policy if testing protocol specific behavior.".. which is assuming that DEFAULT will be used most often and we will simply need to keep upping that number.

Reply:
Yup, agreed. Only the TLS1.2-specific ones should use a numbered policy. And I'd almost say this use case suggests we need a "default_tls12" policy, but I hate that idea and don't want to encourage continued use of TLS1.2 :)

Reply:
Removed the grep_simple_mistakes for Rust code [commit](https://github.com/aws/s2n-tls/pull/4750/commits/e6f3b28adc2fa048c9a25b1a29ee3cc3e7089833).

And we need to pin this to the TLS1.2 security policy, eg. `20240501`, which is already in place.